### PR TITLE
Lowering transpose 

### PR DESCRIPTION
**Pull request by pschang-phy**
### Ticket
https://github.com/tenstorrent/pytorch2.0_ttnn/issues/543

### Problem description
1. `aten.transpose.int` operation not lowering to TTNN
2. tensor with rank > 4 could not support by `ttnn.transpose`

### What's changed
1. Remove `aten.transpose.int` op guarding block list
2. Try do `ttnn.reshape->ttnn.transpose->ttnn.reshape` to do transpose tensor of rank > 4

### CODE REVIEW

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 1]
```diff
@@ -712,7 +712,45 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
...
+                dim1 = (dim1 + rank) % rank
+
+                # directly transpose only support rank <= 4
+                if rank <= 4:
+                    return g.call_function(ttnn.transpose, args=(args[0], dim0, dim1))
+
+                # Try to reshape,
+                #   eg. the original shape [A, B, C, D, E] shape[dim0] = A, shape[dim1] = D
+                #   reshape to [A, (B*C), D, E]
+                originalShape = list(args[0].meta["val"].size())
```
Comment (jerrysky3) [inline torch_ttnn/passes/lowering/to_tt_pass.py:726]: It looks like with the naming convention in this repo this should be `original_shape`, same below

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 2]
```diff
@@ -712,7 +712,45 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
...
+                    newDim0 += 1
+                    newDim1 += 1
+
+                newShape.append(originalShape[dim0])
+
+                if len(originalShape[dim0 + 1 : dim1]) > 0:
+                    newShape.append(np.prod(originalShape[dim0 + 1 : dim1]))
+                    newDim1 += 1
+
+                newShape.append(originalShape[dim1])
```
Comment (jerrysky3) [inline torch_ttnn/passes/lowering/to_tt_pass.py:740]: IIUC here we assume `dim0 < dim1`, maybe need to check and swap if needed

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 3]
```diff
@@ -712,7 +712,45 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
...
-                return g.call_function(ttnn.transpose, args=(args[0], dim0, dim1))
+
+                dim0 = (dim0 + rank) % rank
+                dim1 = (dim1 + rank) % rank
+
+                # directly transpose only support rank <= 4
+                if rank <= 4:
+                    return g.call_function(ttnn.transpose, args=(args[0], dim0, dim1))
+
+                # Try to reshape,
```
Comment (jerrysky3) [inline torch_ttnn/passes/lowering/to_tt_pass.py:723]: This seems to be a workaround. Do we have an issue number can be documented here?

**Review by jerrysky3 — APPROVED** [review]

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 3 above)
Comment (pschang-phy) [reply torch_ttnn/passes/lowering/to_tt_pass.py:723]: I leave a issue to track rank > 4 `ttnn.transpose` issue. https://github.com/tenstorrent/pytorch2.0_ttnn/issues/585,

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 1 above)
Comment (pschang-phy) [reply torch_ttnn/passes/lowering/to_tt_pass.py]: Fixed, refer to 7f572c6eed894451812105f55b72524e501d026e. thanks~

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 2 above)
Comment (pschang-phy) [reply torch_ttnn/passes/lowering/to_tt_pass.py]: Oh yes, It has to check in advance. fixed in 7f572c6eed894451812105f55b72524e501d026e.

**Pull request merged**